sdk/key_vault/src/key.rs

```diff
@@ -0,0 +1,334 @@
+    /// X component of an EC public key.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    x: Option<String>,
+    /// Y component of an EC public key.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    y: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Getters)]
+#[getset(get = "pub")]
+pub struct KeyOperationResult {
+    kid: String,
+    value: String,
+}
+
+/// The signing/verification algorithm identifier
+#[derive(Debug)]
+pub enum JsonWebKeySignatureAlgorithm {
+    ES256,  // ECDSA using P-256 and SHA-256, as described in https://tools.ietf.org/html/rfc7518.
+    ES256K, // ECDSA using P-256K and SHA-256, as described in https://tools.ietf.org/html/rfc7518
+    ES384,  // ECDSA using P-384 and SHA-384, as described in https://tools.ietf.org/html/rfc7518
+    ES512,  // ECDSA using P-521 and SHA-512, as described in https://tools.ietf.org/html/rfc7518
+    PS256, // RSASSA-PSS using SHA-256 and MGF1 with SHA-256, as described in https://tools.ietf.org/html/rfc7518
+    PS384, // RSASSA-PSS using SHA-384 and MGF1 with SHA-384, as described in https://tools.ietf.org/html/rfc7518
+    PS512, // RSASSA-PSS using SHA-512 and MGF1 with SHA-512, as described in https://tools.ietf.org/html/rfc7518
+    RS256, // RSASSA-PKCS1-v1_5 using SHA-256, as described in https://tools.ietf.org/html/rfc7518
+    RS384, // RSASSA-PKCS1-v1_5 using SHA-384, as described in https://tools.ietf.org/html/rfc7518
+    RS512, // RSASSA-PKCS1-v1_5 using SHA-512, as described in https://tools.ietf.org/html/rfc7518
+}
+
+impl Display for JsonWebKeySignatureAlgorithm {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+impl<'a, T: TokenCredential> KeyVaultClient<'a, T> {
+    /// Gets the public part of a stored key.
+    /// The get key operation is applicable to all key types.
+    /// If the requested key is symmetric, then no key material is released in the response.
+    /// This operation requires the keys/get permission.
+    ///
+    /// GET {vaultBaseUrl}/keys/{key-name}/{key-version}?api-version=7.1
+    pub async fn get_key(
+        &mut self,
+        key_name: &str,
+        key_version: &str,
+    ) -> Result<KeyBundle, KeyVaultError> {
+        let uri = Url::parse_with_params(
+            &format!(
+                "{}/keys/{}/{}",
+                self.keyvault_endpoint, key_name, key_version
+            ),
+            &[("api-version", API_VERSION)],
+        )
+        .unwrap();
+        let resp_body = self.get_authed(uri.to_string()).await?;
+        let response = serde_json::from_str::<KeyBundle>(&resp_body)?;
+        Ok(response)
+    }
+
+    /// Creates a signature from a digest using the specified key.
+    /// The SIGN operation is applicable to asymmetric and symmetric keys stored in Azure Key Vault since this operation uses the private portion of the key.
+    /// This operation requires the keys/sign permission.
+    pub async fn sign(
```

Comment:
We could probably accept this for now, but should open a bug. In all other languages' SDKs, we have a separate `CryptographyClient` that takes either a keyId URI or a JsonWebKey (the latter is new that is coming out soon) and performs cryptographic operations on that keyId. We don't do this in the `KeyClient` itself which is meant for key management.

Reply:
Given the temporary nature, `key_name` and `key_version` could stay, but `value` should be `digest` and `alg` should be listed before `digest` and named `algorithm`. See https://github.com/Azure/azure-sdk-for-java/blob/08eadff95aa5a64dcefd8ef58a5bd29d536ab736/sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClient.java#L320.